frontend/lib/framework/error-classes-list.cpp

```diff
@@ -537,4 +537,138 @@ void ErrorProcDefExplicitAnonFormal::write(ErrorWriterBase& wr) const {
+  errorMessage.push_back('.');
+  wr.heading(kind_, type_, loc, errorMessage);
+}
+
+void ErrorCannotAttachPragmas::write(ErrorWriterBase& wr) const {
+  auto loc = std::get<const Location>(info);
+  wr.heading(kind_, type_, loc, "cannot attach pragmas to this statement.");
+}
+
+void ErrorInvalidIndexExpr::write(ErrorWriterBase& wr) const {
+  auto loc = std::get<const Location>(info);
+  wr.heading(kind_, type_, loc, "invalid index expression.");
+}
+
+void ErrorRecordInheritanceNotSupported::write(ErrorWriterBase& wr) const {
+  auto loc = std::get<const Location>(info);
+  wr.heading(kind_, type_, loc,
+             "inheritance is not currently supported for records.");
+  wr.message(
+      "thoughts on what record inheritance should entail can be added to "
+      "https://github.com/chapel-lang/chapel/issues/6851");
+}
+
+void ErrorInvalidNumericLiteral::write(ErrorWriterBase& wr) const {
+  auto loc = std::get<const Location>(info);
+  auto literalStr = std::get<1>(info);
+  auto errMessage = std::get<2>(info);
+  wr.heading(kind_, type_, loc, "error building numeric literal: ", errMessage,
+             ".");
+}
+
+void ErrorMultipleExternalRenaming::write(ErrorWriterBase& wr) const {
+  auto loc = std::get<const Location>(info);
+  wr.heading(
+      kind_, type_, loc,
+      "external symbol renaming can only be applied to one symbol at a time.");
+}
+
+void ErrorPreIncDecOp::write(ErrorWriterBase& wr) const {
+  auto loc = std::get<const Location>(info);
+  auto opSymbol = std::get<1>(info);
+  auto opName = std::get<2>(info);
+  wr.heading(kind_, type_, loc, "'", opSymbol, "' is not a pre-", opName, ".");
+}
+
+void ErrorNewWithoutArgs::write(ErrorWriterBase& wr) const {
+  auto loc = std::get<const Location>(info);
+  wr.heading(kind_, type_, loc,
+             "type in 'new' expression is missing its argument list.");
+}
+
+void ErrorUseImportNeedsModule::write(ErrorWriterBase& wr) const {
+  auto loc = std::get<const Location>(info);
+  auto isImport = std::get<bool>(info);
+  std::string useOrImport = isImport ? "import" : "use";
+  wr.heading(kind_, type_, loc, "'", useOrImport,
+             "' statements must refer to module", (isImport ? "" : " or enum"),
+             " symbols.");
+  wr.message("(e.g., '", useOrImport, " <module>[.<submodule>]*;')");
+}
+
+void ErrorExceptOnlyInvalidExpr::write(ErrorWriterBase& wr) const {
+  auto loc = std::get<const Location>(info);
+  auto limitationKind = std::get<uast::VisibilityClause::LimitationKind>(info);
+  wr.heading(kind_, type_, loc, "incorrect expression in '", limitationKind,
+             "' list, identifier expected.");
+}
+
+void ErrorLabelIneligibleStmt::write(ErrorWriterBase& wr) const {
+  auto loc = std::get<const Location>(info);
+  auto maybeStmt = std::get<const uast::AstNode*>(info);
+  std::string msg = "cannot label";
+  if (maybeStmt) msg = msg + " '" + tagToString(maybeStmt->tag()) + "'";
+  msg += " statement.";
+  wr.heading(kind_, type_, loc, msg);
```

Comment:
Personally, I'd just write this as:
```suggestion
  if (maybeStmt == nullptr) {
    wr.heading(kind_, type_, loc, "cannot label statement.");
  } else {
    wr.heading(kind_, type_, loc, "cannot label '",
                        tagToString(maybeStmt->tag()),"' statement.");
  }
```
In my opinion, the string concatenation operations are kind of awkward here (+ aren't bracelets ifs like this evil?)

Reply:
Braceless `if`s are somewhat evil, though I have started using them occasionally in cases where I don't expect one would want to ever put more than one statement in the `then`. I agree with these changes however.